geospaas/nansat_ingestor/managers.py

```diff
@@ -124,12 +132,17 @@ def get_or_create(self,
             geometry=WKTReader().read(n.get_border_wkt(nPoints=n_points)))[0]
 
         # create dataset
-        ds, created = Dataset.objects.get_or_create(
-            time_coverage_start=n.get_metadata('time_coverage_start'),
-            time_coverage_end=n.get_metadata('time_coverage_end'),
-            source=source,
-            geographic_location=geolocation,
-            **options)
+        ds, created = Dataset.objects.update_or_create(entry_id=entry_id, defaults={
+            'time_coverage_start': n.get_metadata('time_coverage_start'),
+            'time_coverage_end': n.get_metadata('time_coverage_end'),
+            'source': source,
+            'geographic_location': geolocation,
+            'gcmd_location': options["gcmd_location"],
+            'ISO_topic_category': options["ISO_topic_category"],
+            "data_center": options["data_center"],
+            'entry_title': options["entry_title"],
+            'summary': options["summary"]}
+        )
```

Comment:
Is there a particular reason not to keep using `**options`?

Reply:
Since it is the critical part of the code. It is better to explicitly show how the record of dataset is updated by this `update_or_create` command. Actually, the reason is to make it obvious at the maximum level of clarity.